code/game/objects/items/AI_modules.dm

```diff
@@ -523,73 +527,94 @@ AI MODULES
+	return msg
 
 /******************** Robodoctor ****************/
 
 /obj/item/ai_module/core/full/hippocratic
 	name = "'Robodoctor' Core AI Module"
 	law_id = "hippocratic"
 
+
 /******************** Reporter *******************/
 
 /obj/item/ai_module/core/full/reporter
 	name = "'Reportertron' Core AI Module"
 	law_id = "reporter"
 
+
 /****************** Thermodynamic *******************/
 
 /obj/item/ai_module/core/full/thermurderdynamic
 	name = "'Thermodynamic' Core AI Module"
 	law_id = "thermodynamic"
 
 
-/******************Live And Let Live*****************/
+/****************** Live And Let Live *****************/
 
 /obj/item/ai_module/core/full/liveandletlive
 	name = "'Live And Let Live' Core AI Module"
 	law_id = "liveandletlive"
 
-/******************Guardian of Balance***************/
+
+/****************** Guardian of Balance ***************/
 
 /obj/item/ai_module/core/full/balance
 	name = "'Guardian of Balance' Core AI Module"
 	law_id = "balance"
 
+
+/******************** Station Efficiency ******************/
+
 /obj/item/ai_module/core/full/maintain
 	name = "'Station Efficiency' Core AI Module"
 	law_id = "maintain"
 
+
+/******************** Peacekeeper ******************/
+
 /obj/item/ai_module/core/full/peacekeeper
 	name = "'Peacekeeper' Core AI Module"
 	law_id = "peacekeeper"
 
-// Bad times ahead
 
-/obj/item/ai_module/core/full/damaged
-		name = "damaged Core AI Module"
-		desc = "An AI Module for programming laws to an AI. It looks slightly damaged."
-
-/obj/item/ai_module/core/full/damaged/install(datum/ai_laws/law_datum, mob/user)
-	laws += generate_ion_law()
-	while (prob(75))
-		laws += generate_ion_law()
-	..()
-	laws = list()
-
-/******************H.O.G.A.N.***************/
+/****************** H.O.G.A.N. ***************/
 
 /obj/item/ai_module/core/full/hulkamania
 	name = "'H.O.G.A.N.' Core AI Module"
 	law_id = "hulkamania"
 
 
-/******************Overlord***************/
+/****************** Overlord ***************/
 
 /obj/item/ai_module/core/full/overlord
 	name = "'Overlord' Core AI Module"
 	law_id = "overlord"
+
+
+/******************** Damaged ******************/
+
+/obj/item/ai_module/core/full/damaged
+		name = "damaged Core AI Module"
+		desc = "An AI Module for changing the programming of an AI's core laws. It looks slightly damaged."
+
+/obj/item/ai_module/core/full/damaged/install(datum/ai_laws/law_datum, mob/user)
+	laws += generate_ion_law()
+	while (prob(75))
+		laws += generate_ion_law()
```

Comment:
Can you limit how many laws this can make?

Reply:
I think I'd rather it just `pickweight` something where 2 laws is the most common, and the highest is something like 5 with a very low chance.

Reply:
this isn't my code

Reply:
Then I guess we'll make it better later.